src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -162,8 +162,42 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                     arg = PSObject.ToStringParser(context, ParserOps.Current(null, list));
                 }
 
+#if UNIX
+                Collection<string> resolvedPaths = new Collection<string>();
+#endif
                 if (!String.IsNullOrEmpty(arg))
                 {
+                    // This implements globbing of arguments given to native tools on UNIX.
+                    // Conversely, Windows tools implement their own globbing,
```

Comment:
As I said, the motivation example are tools that exists on both platforms, like git, gcc and so on.